File: exts/jphp-gui-game-ext/src/main/java/org/develnext/jphp/ext/game/classes/UXSpriteView.java
package org.develnext.jphp.ext.game.classes;

import org.develnext.jphp.ext.game.GameExtension;
import org.develnext.jphp.ext.game.support.Sprite;
import org.develnext.jphp.ext.game.support.SpriteView;
import org.develnext.jphp.ext.javafx.classes.UXCanvas;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Nullable;
import php.runtime.annotation.Reflection.Property;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Reflection.Namespace(GameExtension.NS)
public class UXSpriteView extends UXCanvas<SpriteView> {
    interface WrappedInterface {
        @Property @Nullable Sprite sprite();
        @Property("animated") boolean animationEnabled();
        @Property String animationName();
        @Property int animationSpeed();

        @Property int frame();
        @Property int frameCount();

        @Property boolean flipX();
        @Property boolean flipY();

        void play(String animation);
        void play(String animation, int speed);
        void playOnce(String animation);
        void playOnce(String animation, int speed);

        void pause();
        boolean isPaused();
        void resume();

        void update(long now);
    }

    public UXSpriteView(Environment env, SpriteView wrappedObject) {
        super(env, wrappedObject);
    }

    public UXSpriteView(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public void __construct() {
        __wrappedObject = new SpriteView();
    }

    @Signature
    public void __construct(@Nullable Sprite sprite) {
        __wrappedObject = new SpriteView(sprite);
    }
}

File: exts/jphp-gui-game-ext/src/main/resources/JPHP-INF/sdk/php/game/UXSpriteView.php
<?php
namespace php\game;

use php\gui\UXCanvas;

/**
 * Class UXSpriteView
 * @package php\game
 * @packages game, javafx
 */
class UXSpriteView extends UXCanvas
{
    /**
     * Спрайт.
     * @var UXSprite
     */
    public $sprite = null;

    /**
     * Анимирован.
     * @var bool
     */
    public $animated = false;

    /**
     * Анимация.
     * @var null|string
     */
    public $animationName = null;

    /**
     * Скорость анимации.
     * @var int
     */
    public $animationSpeed = 12;

    /**
     * Кадр спрайта.
     * @var int
     */
    public $frame = -1;

    /**
     * Количество кадров.
     * @readonly
     * @var int
     */
    public $frameCount = 0;

    /**
     * Отразить спрайт по X.
     * @var bool
     */
    public $flipX = false;

    /**
     * Отразить спрайт по Y.
     * @var bool
     */
    public $flipY = false;

    /**
     * UXSpriteView constructor.
     * @param UXSprite $sprite
     */
    public function __construct(UXSprite $sprite = null)
    {
    }

    /**
     * Проигрывать анимацию циклично.
     * @param string $animation
     * @param int $speed (optional)
     */
    public function play($animation, $speed = -1)
    {
    }

    /**
     * Проиграть анимацию единожды.
     * @param string $animation
     * @param int $speed
     */
    public function playOnce($animation, $speed = -1)
    {
    }

    public function pause()
    {
    }

    public function resume()
    {
    }

    /**
     * @return bool
     */
    public function isPaused()
    {
    }
}